### ISSUE TITLE
SL Review: Add Department ID

### ISSUE THREAD

**Issue by Stella-Archer** [bcgov/digital-journeys#1280]
**User Story**
“As a BI expert I want Department ID of nominees associated with each SL Review submission so I can track and find each nomination and create accurate reliable reports” 

**Tasks**
Steps needed to achieve this feature. 

**Acceptance Criteria**
department ID is identified on each submission to the ODS

**Comment by Andrew-Vargas-bcgov** [comment]
Need to add deptId to the employee lookup and also the db.

**Comment by ayushdamani** [comment]
Tested and confirmed working ✔️

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBcU1hIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--be433160f6fbd7f89221e34c20db182556a24b03/image.png)

**Comment by Stella-Archer** [comment]
looks good in test - is it showing up in for the data people?
PO'd

**Comment by Abuchana** [comment]
@mikeshi-bcgov Looks good in test - is it showing up in the data
PO'd

**Comment by bhumin-fw** [comment]
@mikeshi-bcgov Added one new submission on TEST. Name of employee Ayush Damani and submitted by Bhumin Patel.

**Comment by mikeshi-bcgov** [comment]
@bhumin-fw received record applicationid =2939, but no DepartmentID value

**Comment by bhumin-fw** [comment]
@mikeshi-bcgov 
The param we are sending for departmentID is deptid

e.g (recent submission):
deptid :  "100-3539"

**Comment by mikeshi-bcgov** [comment]
@bhumin-fw can you please share with me the submission with JSON format?

**Comment by bhumin-fw** [comment]
@mikeshi-bcgov 
Sent full object in JSON form on your email. Please do let me know if you haven't received it.

**Comment by mikeshi-bcgov** [comment]
@bhumin-fw ok, got wrong field name, just aligned with yours "deptid", please test again, thx

**Comment by mikeshi-bcgov** [comment]
@bhumin-fw confirm received, looks good, thx

**Comment by iman-jamali-fw** [comment]
@Stella-Archer 
Ready to be tested on TEST. Please also check with @mikeshi-bcgov to make sure the date is received at ODS (please share applicationId of your submission with him).

**Comment by mikeshi-bcgov** [comment]
Today we received 1 submission applicationid =9810, deptid =100-3547, hope it is the one tested

**Comment by iman-jamali-fw** [comment]
@ayushdamani 
I don't see an SL submission on the test environment.

For testing this ticket and making sure it's production ready, could you please do another round of testing with instruction below:
1. Submit the form
2. Go to the `Submitted Forms` tab and edit your submission, click `Resubmit for validation`. Make sure it works and check the application in the Camunda making sure there is no error for your submission. 
3. Go back to your submission and this time click `Final Submission`. Make sure it goes through and wait for email to show up in your inbox.
4. Verify the attachment in the email, make sure pdf has the new `depId` filed. Also, verify if the first two sections in the PDF (_About the senior leadership review process_  and _Before you begin_) are supposed to show up there.
5. Finally take note of your submission's `applicationId` and check that with @mikeshi-bcgov - you can get the `applicationId` when opening your submission in the network tab or find it in the Camunda)

Let me know if need anything else that helps with your testing :)

**Comment by iman-jamali-fw** [comment]
@mikeshi-bcgov
There will a special case where an SL user starts their application and will send (resubmit/final submit) their application after `depId` field is added to PROD. In that case, `depId` will be `null`. I handle that case on the frontend side by making that field is not required. Could you please make sure this field is nullable on the ODS side - if not already? Thank you

**Comment by ayushdamani** [comment]
Thank you for the instructions! Everything worked great, but I was just not able to find the applicationId.

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBcnMwIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--943b85865722cc6f72a63d85a1ec018c98cb20b0/Screenshot%202023-08-24%20052406.png)

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBcjQwIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--734a9bf5ef9091ed58f594b9946c63d5170211f0/Screenshot%202023-08-24%20052448.png)

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBcjgwIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--ee65266b976ddea9d33442632326815e711e58e5/Screenshot%202023-08-24%20052525.png)

**Comment by iman-jamali-fw** [comment]
@ayushdamani 
To find applicationId of a submission, please open that submission under `Submitted Forms` and then open the network tab of your browser (you can right-click and select `inspect`) and follow the screenshot below:
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBa2MyIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--29829ca70fa0d499c8662888d9ecdc25579cb824/Screenshot%202023-08-24%20at%208.43.23%20AM.png)

**Comment by ayushdamani** [comment]
I really appreciate this, @iman-jamali-fw ! Thank you!
I was able to follow the steps you mentioned.

**Comment by mikeshi-bcgov** [comment]
We received 2 submissions this morning, 2964 and 9827, didn't receive 2965.

**Comment by iman-jamali-fw** [comment]
Thanks @mikeshi-bcgov

I see 2965 is already deleted. @ayushdamani Did you by any chance delete that submission?

**Comment by ayushdamani** [comment]
Hi Iman,
No, I have not deleted that submission. Would you like me to test with another form?

**Comment by MeghanStothers** [comment]
Tested by Dj team this morning - confirm that dept ID appears in the form as well as the pdfs. Over to Jodi for her team to test. If everything looks good, we can close this ticket

**Comment by MeghanStothers** [comment]
Tested with Jodi who ran a report - she did not see the dept ID.  @Andrew-Vargas-bcgov Andrew to connect with Sarah Miles to make sure the ID shows on a report

**Comment by mikeshi-bcgov** [comment]
@iman-jamali-fw please let us know when you will deploy this feature to prod, so we will deploy changes accordingly. We have CR every Thursday.

**Comment by iman-jamali-fw** [comment]
@mikeshi-bcgov
Hi Mike
Yes! we deployed this to PROD and new submissions today and on should have a `deptId` field.

**Comment by mikeshi-bcgov** [comment]
@iman-jamali-fw Thanks, we will deploy the change to Prod tomorrow morning.

**Comment by MeghanStothers** [comment]
Tested this one with the PSA dev team in prod and see the dept ID field on the form and in the PDF - final steps is to confirm with Sarah Miles  that the dept ID shows on the reporting.  We tested with Dodi yesterday and the dept ID field did not show on the report she ran. Andrew to confirm this filed is included in the report by connecting with Sarah to confirm

**Comment by Abuchana** [comment]
@JodiLittle @sarah.miles@gov.bc.ca any new submissions today will have dept. ID, anything else submitted before today will not unless re-submitted. If necessary, reports can be manually updated if the BI team is provided the employees associated with the dept. IDs. Please contact Mike Shi directly.

**Comment by iman-jamali-fw** [comment]
@mikeshi-bcgov
Yeah, seems like some of the new submissions haven't picked up the `deptid`. Let me look into this.

**Comment by iman-jamali-fw** [comment]
@mikeshi-bcgov
I looked into all the submissions that were sent today without `deptid`. They were started before the fix was deployed to the PROD, and today were simply resubmitted. There are two dates in the submissions,`submissionDate` that indicates when it was first submitted and `validationDate` that updates when it is resubmitted.

- Submissions that were started before Aug 28, won't have `deptid` unless manually added.
- Submissions that were sent on Aug 28, will get `deptid` once resubmitted. This is because `deptid` was added to DGJ but wasn't deployed at ODS.
- Submissions that are sent on Aug 29 and later, will have `depid`.
